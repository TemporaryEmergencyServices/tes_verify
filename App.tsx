import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';

import useCachedResources from './hooks/useCachedResources';
import useColorScheme from './hooks/useColorScheme';
import Navigation from './navigation';


//The following are for redux
import {createStore, applyMiddleware} from 'redux'
import {Provider} from 'react-redux'
import thunkMiddleware from 'redux-thunk'
// import reducer from './reducers'
import rootReducer from './reducers'

const middleware = applyMiddleware(thunkMiddleware)
export const store = createStore(rootReducer, middleware)
// const store = createStore(rootReducer)
//End redux

export default function App() {
  const isLoadingComplete = useCachedResources();
  const colorScheme = useColorScheme();

  if (!isLoadingComplete) {
    return null;
  } else {
    return (
      //Provider wrapper and store area for redux
      <Provider store={store}>
        <SafeAreaProvider>
            <Navigation colorScheme={colorScheme} />
            <StatusBar />
        </SafeAreaProvider>
      </Provider>
    );
  }
}