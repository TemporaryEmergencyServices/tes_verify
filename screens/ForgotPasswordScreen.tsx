import firebase from '../firebase.js'
import React, { useState } from 'react';
import { StyleSheet, Text, View, TextInput, TouchableOpacity } from 'react-native';


import { bindActionCreators } from 'redux';
import {Dispatch} from 'redux';
import { connect } from 'react-redux';
import { updateEmail, updatePassword, signup } from '../actions/user';
/*
 * Code is loosely based on the following tutorials: 
 * https://reactnativemaster.com/react-native-login-screen-tutorial
 * https://heartbeat.fritz.ai/how-to-implement-forgot-password-feature-in-react-native-and-firebase-app-890b572d9759
 * */

//gives warning for navigation - this goes away if you uncomments the 'noImplicitAny' line from tsconfig
//unsure of other impacts of having that line, so uncommenting may be a bad idea
function ForgotPasswordScreen({  navigation  }) {
  const [emailState, setEmailState] = useState('')
  const goToSignIn = () => navigation.replace('SignInScreen')
  const handlePasswordReset = () => {
    firebase.auth()
      .sendPasswordResetEmail(emailState)
  }
  return (
    <View style={styles.container}>
      <Text style={styles.logo}>TES Verify</Text>
      <View style={styles.inputView} >
        <TextInput  
          autoCapitalize="none"
          style={styles.inputText}
          placeholder="Email..." 
          placeholderTextColor="white"
          onChangeText={text => setEmailState(text)}/>
      </View>
      <TouchableOpacity style={styles.loginBtn} onPress={handlePasswordReset}>
        <Text style={styles.signUpText} >Reset Password</Text>
      </TouchableOpacity>
      <TouchableOpacity onPress = {goToSignIn}>
        <Text style={styles.returnToSignInText}>Return to Sign In</Text>
      </TouchableOpacity>
    </View>
  );
  
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFFFFF',
    alignItems: 'center',
    justifyContent: 'center',
  },
  logo:{
    fontWeight:"bold",
    fontSize:50,
    color:"#1C5A7D",
    marginBottom:40
  },
  inputView:{
    width:"80%",
    backgroundColor:"#2B2E32",
    borderRadius:25,
    height:50,
    marginBottom:20,
    justifyContent:"center",
    padding:20
  },
  inputText:{
    height:50,
    color:"white"
  },
  forgot:{
    color:"white",
    fontSize:11
  },
  loginBtn:{
    width:"80%",
    backgroundColor:"#E11383",
    borderRadius:25,
    height:50,
    alignItems:"center",
    justifyContent:"center",
    marginTop:40,
    marginBottom:10
  },
  createAccountBtn:{
    width:"80%",
    backgroundColor:"#E11383",
    borderRadius:25,
    height:50,
    alignItems:"center",
    justifyContent:"center",
    marginTop:10,
    marginBottom:10
  },
  returnToSignInText:{
    color:"black"
  },
  signUpText:{
    color:"white",
    fontSize: 18,
    fontWeight: "bold"
  }
});


//Redux logic that doesn't work (probably because this is a function component and not a class component?)
const mapDispatchToProps = (dispatch: Dispatch)=>{
  return bindActionCreators({ updateEmail, updatePassword, signup }, dispatch);
}

const  mapStateToProps = (state: any) => {
  return {
      user: state.user
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ForgotPasswordScreen);
