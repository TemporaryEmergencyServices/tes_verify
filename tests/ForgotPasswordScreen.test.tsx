import React from "react";
import renderer from "react-test-renderer";
import ForgotPasswordScreen from "../screens/ForgotPasswordScreen"
import { Provider } from 'react-redux'
import { store } from '../App'

jest.useFakeTimers()

describe("<ForgotPasswordScreen />", () => {
    it('renders correctly', () => {
        const navigation={}
        const tree = renderer.create(<Provider store={store} ><ForgotPasswordScreen navigation={navigation}/></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});