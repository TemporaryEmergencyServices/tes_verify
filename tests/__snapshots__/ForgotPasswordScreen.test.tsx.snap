// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPasswordScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1C5A7D",
        "fontSize": 50,
        "fontWeight": "bold",
        "marginBottom": 40,
      }
    }
  >
    TES Verify
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#2B2E32",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 20,
        "padding": 20,
        "width": "80%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Email..."
      placeholderTextColor="white"
      rejectResponderTermination={true}
      style={
        Object {
          "color": "white",
          "height": 50,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E11383",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 40,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Reset Password
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
        }
      }
    >
      Return to Sign In
    </Text>
  </View>
</View>
`;
